docs/manuals/core/entities/data-node-mgt.md

```diff
@@ -165,4 +144,577 @@ writes it on the data node:
+        data_node.write(data)
+        ```
+
+        results in:
+
+        ```json
+        [
+            {"__type__": "dataclass-SaleRow", "__value__": {"date": "12/24/2018", "nb_sales": 1550}},
+            {"__type__": "dataclass-SaleRow", "__value__": {"date": "12/25/2018", "nb_sales": 2315}},
+            {"__type__": "dataclass-SaleRow", "__value__": {"date": "12/26/2018", "nb_sales": 1832}},
+        ]
+        ```
+
+        The read method will return a list of SaleRow objects as `data` when written.
+
+You can also specify custom JSON _**encoder**_ and _**decoder**_ to handle different data types. Check out [JSON Data Node
+configuration](../config/data-node-config.md#json) for more details on how to config custom JSON _**encoder**_ and _**decoder**_.
+
+## Mongo collection
+
+When read from a Mongo collection data node, Taipy will return a list of objects as instances of a document class defined by _**custom_document**_.
+
+When write data to a Mongo collection data node, the `MongoCollectionDataNode.write()^` method takes a list of objects as instances of a document class
+defined by _**custom_document**_ as the input.
+
+By default, Mongo collection data node uses `taipy.core.DefaultCustomDocument` as the document class. A `DefaultCustomDocument` can have any attribute,
+however, the type of the value should be supported by MongoDB, including but not limited to:
+
+- Boolean, integers, and floating-point numbers.
+- String.
+- Object (embedded document object).
+- Arrays − arrays or list or multiple values.
+
+Check out [MongoDB supported data types](https://www.mongodb.com/docs/manual/reference/bson-types/) for more details.
+
+!!! example "Read and write from a Mongo collection data node using default document class"
+
+    ```python
+    from taipy.core import DefaultCustomDocument
+
+    data = [
+        DefaultCustomDocument(date="12/24/2018", nb_sales=1550),
+        DefaultCustomDocument(date="12/25/2018", nb_sales=2315),
+        DefaultCustomDocument(date="12/26/2018", nb_sales=1832),
+    ]
+    data_node.write(data)
+    ```
+
+    will write 3 documents to MongoDB:
+
+    ```json
+    [
+        {"_id": ObjectId("634cd1b3383279c68cee1c21"), "date": "12/24/2018", "nb_sales": 1550},
+        {"_id": ObjectId("634cd1b3383279c68cee1c22"), "date": "12/25/2018", "nb_sales": 2315},
+        {"_id": ObjectId("634cd1b3383279c68cee1c23"), "date": "12/26/2018", "nb_sales": 1832},
+    ]
+    ```
+
+    The read method will return a list of DefaultCustomDocument objects, including "_id" attribute.
+
+You can also specify custom document class to handle specific attribute, encode and decode data when reading and writing to the Mongo collection.
+
+Check out [Mongo collection Data Node configuration](../config/data-node-config.md#mongo-collection) for more details on how to config a custom
+document class.
+
+## In memory
+
+In memory read / write example
+
+
+## Generic
+
+Generic read / write example
+
+
+# Filter read results
```

Comment:
Can you check that the filter method is working for any kind of data node. I am not sure anymore at this point.

Reply:
For now, the filter method only work when the return type of read method is DataFrame, or List. So when the expose type is numpy, a custom class, or MongoCollectionDataNode, this will not work.